### PR TITLE
Concurrency and rate management docs

### PR DESCRIPTION
**Pull request by psavkar**

### CODE REVIEW

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Managing Concurrency and Throttling
+
+Pipedream makes it easy to manage the concurrency and rate at which events from a source trigger your workflow code using execution controls.
+
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:4]: Can you add a `[[toc]]` section here?

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- Up to 100 events will be queued per workflow for free and pro accounts
+- Team/Enterprise accounts may have custom limits. If you need a larger queue size, please contact Pipedream
+
+**IMPORTANT:** If the number of events emitted to a workflow exceeds the queue size, events will be lost. If that happens, an error message will be displayed in the event list of your workflow and your global error workflow will be triggered.
+
+For more context on this feature and technical details, check out our **engineering blog post**.
+
+## Where Do I Manage Concurrency and Throttling?
+
+Concurency and throttling can be managed in the **Execution Controls** section of your **Workflow Settings**. Event queues are currently supported for any workflow that is triggered by an event source. Event queues are not currently supported for native workflow triggers (native HTTP, cron, SDK and email).
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:44]: "Concurrency"

Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:44]: Can you link to the event sources and native workflows trigger sections from this section so newcomers can see exactly the distinction between those?

https://docs.pipedream.com/event-sources/
https://docs.pipedream.com/workflows/steps/triggers/

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Events emitted from a source to a workflow are placed in a queue, and Pipedream triggers your workflow with events from the queue based on your concurrency and rate limit settings. These settings may be customized per workflow (so the same events may be processed at different rates by different workflows).
+
+![image-20201027145847752](images/image-20201027145847752.png)
+
+The maximum number of events Pipedream will queue per workflow depends on your account type.
+
+- Up to 100 events will be queued per workflow for free and pro accounts
+- Team/Enterprise accounts may have custom limits. If you need a larger queue size, please contact Pipedream
+
+**IMPORTANT:** If the number of events emitted to a workflow exceeds the queue size, events will be lost. If that happens, an error message will be displayed in the event list of your workflow and your global error workflow will be triggered.
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:38]: it's a Queue Size Exceeded limit right? We should probably specify that and add that error to https://docs.pipedream.com/errors/#pipedream-internal-errors

Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:38]: Link to global error workflow docs

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## How It Works
+
+Events emitted from a source to a workflow are placed in a queue, and Pipedream triggers your workflow with events from the queue based on your concurrency and rate limit settings. These settings may be customized per workflow (so the same events may be processed at different rates by different workflows).
+
+![image-20201027145847752](images/image-20201027145847752.png)
+
+The maximum number of events Pipedream will queue per workflow depends on your account type.
+
+- Up to 100 events will be queued per workflow for free and pro accounts
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:35]: Link to plans

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# Managing Concurrency and Throttling
+
+Pipedream makes it easy to manage the concurrency and rate at which events from a source trigger your workflow code using execution controls.
+
+## Why Is It Important?
+
+Workflows listen for events and execute as soon as they are triggered. While this behavior is expected for many use cases, there can be unintended consequences. 
+
+**Concurrency**
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:9]: Maybe use h3 tags (`###`) so these sections can be linked to / show up in ToC?

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+![image-20201027132901691](./images/image-20201027132901691.png)
+
+You can avoid race conditions like this by limiting workflow concurrency to a single "worker". What this means is that only one event will be processed at a time, and the next event will not start processing until the first is complete (unprocesssed events will maintained in a queue and processed by the workflow in order). The following diagram illustrates how the events in the last diagram would executed if concurrency was limited to a single worker. 
+
+![image-20201027133308888](./images/image-20201027133308888.png)
+
+While the first example resulted in only two rows of data in Google Sheets, this time data for all three events are recorded to three separate rows.
+
+**Throttling**
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:23]: Maybe use h3 tags (`###`) so these sections can be linked to / show up in ToC?

File: docs/docs/workflows/events/concurrency-and-throttling/README.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Why Is It Important?
+
+Workflows listen for events and execute as soon as they are triggered. While this behavior is expected for many use cases, there can be unintended consequences. 
+
+**Concurrency**
+
+Without restricting concurrency, events can be processed in parallel and there is no guarantee that they will execute in the order in which they were received. This can cause race conditions. 
+
+For example, if two workflow events try to add data to Google Sheets simultaneously, they may both attempt to write data to the same row. As a result, one event can overwrite data from another event. The diagram below illustrates this example -- both `Event 1` and `Event 2` attempt to write data to Google Sheets concurrently -- as a result, they will both write to the same row and the data for one event will be overwritten and lost (and no error will be thrown). We observed this scenario resulted in data loss approximately 20% of the time with high volume workflows.
```
Comment (dylburger) [inline docs/docs/workflows/events/concurrency-and-throttling/README.md:13]: Replace `--` with em dash (`—`). You can insert with Shift + Option + `-`

File: docs/docs/workflows/events/concurrency-and-throttling/README.md (hunk 1 above)
Comment (psavkar) [reply docs/docs/workflows/events/concurrency-and-throttling/README.md:4]: done

File: docs/docs/workflows/events/concurrency-and-throttling/README.md (hunk 4 above)
Comment (psavkar) [reply docs/docs/workflows/events/concurrency-and-throttling/README.md:37]: done

File: docs/docs/workflows/events/concurrency-and-throttling/README.md (hunk 3 above)
Comment (psavkar) [reply docs/docs/workflows/events/concurrency-and-throttling/README.md:40]: done

**Pull request merged**